src/test_application/OSPMagnum.cpp

```diff
@@ -159,71 +161,195 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
-            {{0, (int) Key_t::Q, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-    rUserInput.config_register_control("vehicle_roll_rt", true,
-            {{0, (int) Key_t::E, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-
-    // Set throttle max to Z
-    rUserInput.config_register_control("vehicle_thr_max", false,
-            {{0, (int) Key_t::Z, VarTrig_t::PRESSED, false, VarOp_t::OR}});
-    // Set throttle min to X
-    rUserInput.config_register_control("vehicle_thr_min", false,
-            {{0, (int) Key_t::X, VarTrig_t::PRESSED, false, VarOp_t::OR}});
-    // Set self destruct to LeftCtrl+C or LeftShift+A
-    rUserInput.config_register_control("vehicle_self_destruct", false,
-            {{0, (int) Key_t::LeftCtrl, VarTrig_t::HOLD, false, VarOp_t::AND},
-             {0, (int) Key_t::C, VarTrig_t::PRESSED, false, VarOp_t::OR},
-             {0, (int) Key_t::LeftShift, VarTrig_t::HOLD, false, VarOp_t::AND},
-             {0, (int) Key_t::A, VarTrig_t::PRESSED, false, VarOp_t::OR}});
-
-    // Camera and Game controls, handled in DebugCameraController
-
-    // Switch to next vehicle
-    rUserInput.config_register_control("game_switch", false,
-            {{0, (int) Key_t::V, VarTrig_t::PRESSED, false, VarOp_t::OR}});
-
-    // Set UI Up/down/left/right to arrow keys. this is used to rotate the view
-    // for now
-    rUserInput.config_register_control("ui_up", true,
-            {{osp::sc_keyboard, (int) Key_t::Up, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-    rUserInput.config_register_control("ui_dn", true,
-            {{osp::sc_keyboard, (int) Key_t::Down, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-    rUserInput.config_register_control("ui_lf", true,
-            {{osp::sc_keyboard, (int) Key_t::Left, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-    rUserInput.config_register_control("ui_rt", true,
-            {{osp::sc_keyboard, (int) Key_t::Right, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-
-    rUserInput.config_register_control("ui_rmb", true,
-            {{osp::sc_mouse, (int) Mouse_t::Right, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-
-    rUserInput.config_register_control("debug_planet_update", false,
-            {{0, (int) Key_t::LeftCtrl, VarTrig_t::HOLD, false, VarOp_t::AND},
-             {0, (int) Key_t::One, VarTrig_t::PRESSED, false, VarOp_t::OR}});
+	// Configure Controls
+
+	// It should be pretty easy to write a config file parser that calls these
+	// functions.
+
+	// It should be pretty easy to write a config file parser that calls these
+	// functions.
+
+	//Load toml
+	auto data = toml::parse("settings.toml");
+	std::cout << "found file " << std::endl;
+
+	using namespace osp;
+
+	using Key_t = OSPMagnum::KeyEvent::Key;
+	using Mouse_t = OSPMagnum::MouseEvent::Button;
+	using VarOp_t = ButtonVarConfig::VarOperator;
+	using VarTrig_t = ButtonVarConfig::VarTrigger;
+
+	UserInputHandler& rUserInput = rOspApp.get_input_handler();
+	try {
```

Comment:
This is the first time I've seen exceptions in this project. I'd need an expert opinion on this.